content/en/docs/concepts/overview/kubernetes-api.md

```diff
@@ -82,18 +82,42 @@ packages that define the API objects.
 
 ### OpenAPI V3
 
-{{< feature-state state="alpha"  for_k8s_version="v1.23" >}}
+{{< feature-state state="beta"  for_k8s_version="v1.24" >}}
 
-Kubernetes v1.23 offers initial support for publishing its APIs as OpenAPI v3; this is an
-alpha feature that is disabled by default.
-You can enable the alpha feature by turning on the
+Kubernetes v1.24 offers beta support for publishing its APIs as OpenAPI v3; this is a
+beta feature that is enabled by default.
+You can disable the beta feature by turning off the
 [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) named `OpenAPIV3`
 for the kube-apiserver component.
 
-With the feature enabled, the Kubernetes API server serves an
-aggregated OpenAPI v3 spec per Kubernetes group version at the
-`/openapi/v3/apis/<group>/<version>` endpoint. Please refer to the
-table below for accepted request headers.
+A discovery endpoint `/openapi/v3` is provided to see a list of all
+group/versions available. This endpoint only returns JSON. These group/versions
+are provided in the following format:
+```
+{
+  "paths": [{
+    "$group_version": {
+      "serverRelativeURL": "$relative_url?hash=$current_resource_hash"
+	},
+	"apps/v1": {
+	  "serverRelativeURL": "/openapi/v3/apis/apps/v1?hash=ha5hha5hha5hha5h"
+	}
+  }]
+}
+```
+
+The hash is provided as part of the URL so that the response from the server can
+be cached indefinitely. The proper HTTP caching headers are also set by the
+apiserver for that purpose (`Expire` to 1 year in the future, and
+`Cache-control` to `immutable`).
+
+The Kubernetes API server serves an aggregated OpenAPI v3 spec per Kubernetes
+group version at the `/openapi/v3/apis/<group>/<version>?hash=$hash` endpoint.
+If the hash is not provided, a redirect will be returned to the most recent
+version of the end-point with the hash, so that caching libraries can cache the
+proper version of the file.
```

Comment:
```suggestion
You can also request the OpenAPI data without specifying a hash in the query
string, and in that case the API server sends a redirect response with the URI
of the current API data.
```
I didn't quite understand under what circumstances a client might discover or
know the OpenAPI URL that doesn't include a hash - can we explain that?

Reply:
Yes, there are merely two cases (describing for context):
1. Someone tries to figure out how to build the path, but they don't have the hashtag (in which case we actually don't redirect, we send the full version without disabled caching headers)
3. a URL/hash becomes obsolete, in which case we redirect to the newer version of the URL.

I'll try to rephrase that better.